src/os/source/vectorsource.js

```diff
@@ -416,6 +416,12 @@ os.source.Vector = function(opt_options) {
    */
   this.hasModifications = false;
 
+  /**
+   * Mappings for the source
+   * @type {Object<string, *>}
+   */
+  this.mappingManagers = {};
```

Comment:
Do/will Mappings only work on VectorSource data? Would it be better to move this storage into MappingManagers and make it a singleton?

Reply:
Thank you Alyssa for explaining where this is going.

So, as a follow on, please create a `SourceMappingManager` that stores MappingManagers by sourceId > MappingManagerTypeEnum > then the MappingManager, e.g. {'12345': {'ellipseMMKey': new EllipseMappingManager()}} and its own getInstance(). That way, this won't add much of anything to VectorSource, which is exported to be shared among windows.

For now, just update the above to use the MappingManagerTypeEnum (i.e. ELLIPSE_MAPPING_KEY) instead of the SourceId so it'll be a smaller change to move this out later.

Reply:
I would personally prefer to keep this out of VectorSource in the initial implementation instead of adding it with the assumption it will be removed later. VectorSource is bloated enough as-is, and it doesn't seem this belongs here.

Reply:
I've added a `SourceMappingManager` that could take care of this. The only other question is if I should create an `AbstractMappingManager` that `EllipseMappingManager` could implement. And also there might be a more ideal way to store mappingmanagers in the new file so let me know if you guys have notes on that.